tests/acceptance/components/readonly-test/edit-form-readonly-test.js

```diff
@@ -0,0 +1,251 @@
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $textarea = Ember.$('.not-in-groupedit .flexberry-textarea');
+    let $textareaInput = $textarea.children('textarea');
+    assert.strictEqual(Ember.$.trim($textareaInput.attr('readonly')), 'readonly', 'Textarea is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($textareaInput.attr('readonly')), '', 'Textarea don\'t readonly');
+    });
+  });
+});
+
+test('flexberry-datepicker on readonly editform', (assert) => {
+  assert.expect(4);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $datepicker = Ember.$('.not-in-groupedit .flexberry-datepicker');
+    let $datepickerInput = $datepicker.children('input');
+    assert.strictEqual(Ember.$.trim($datepickerInput.attr('readonly')), 'readonly', 'Time is readonly');
+
+    let $button = $datepicker.children('.calendar');
+    assert.strictEqual($button.hasClass('link'), false, 'Datepicker hasn\'t link');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($datepickerInput.attr('readonly')), '', 'Time don\'t readonly');
+      assert.strictEqual($button.hasClass('link'), true, 'Datepicker has link');
+    });
+  });
+});
+
+test('flexberry-simpledatetime on readonly editform', (assert) => {
+  assert.expect(2);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $simpledatetime = Ember.$('.not-in-groupedit .flexberry-simpledatetime');
+    let $simpledatetimeInput = $simpledatetime.children('input');
+    assert.strictEqual(Ember.$.trim($simpledatetimeInput.attr('readonly')), 'readonly', 'Time is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual(Ember.$.trim($simpledatetimeInput.attr('readonly')), '', 'Time don\'t readonly');
+    });
+  });
+});
+
+test('flexberry-dropdown on readonly editform', (assert) => {
+  assert.expect(2);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $dropdown = Ember.$('.not-in-groupedit .flexberry-dropdown');
+    assert.strictEqual($dropdown.hasClass('disabled'), true, 'Dropdown is readonly');
+
+    controller.set('readonly', false);
+    Ember.run.scheduleOnce('afterRender', () => {
+      assert.strictEqual($dropdown.hasClass('disabled'), false, 'Enumeration don\'t readonly');
+    });
+  });
+});
+
+test('flexberry-file on readonly edit form', (assert) => {
+  assert.expect(7);
+
+  visit(path);
+  andThen(() => {
+    let controller = app.__container__.lookup('controller:' + currentRouteName());
+    let $file = Ember.$('div.input input.flexberry-file-filename-input');
```

Comment:
You forget about '.not-in-groupedit' class, so now you are selecting both in same time (component simply placed on  the form and those one that is placed inside groupedit).
Selectors like 'div.input input' are useless, you can select input like that: `let $fileNameInput = Ember.$('.not-in-groupedit .flexberry-file .flexberry-file-filename-input');` or like that: `let $flexberryFile = Ember.$('.not-in-groupedit .flexberry-file'); let $fileNameInput = Ember.$(' .flexberry-file-filename-input', $flexberryFile)` same selection ways are available for component's buttons and other markup.